dotnet/Gherkin/Parser.cs

```diff
@@ -75,8 +75,20 @@ public enum RuleType
     [System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
     public partial class Parser<T>
     {
+        public Parser()
+            : this(new AstBuilder<T>())
+        {
+        }
+
+        public Parser(AstBuilder<T> astBuilder)
+        {
+            this.AstBuilder = astBuilder;
+        }
+
         public bool StopAtFirstError { get; set;}
 
+        public AstBuilder<T> AstBuilder { get; }

```

Comment:
That depends on the version of C# :-) C# 6 (introduced with Visual Studio 2015) allows me to write a readonly property like that. If you feel that this would cause problems for the build servers or many other contributors than I will change it to a `readonly` field.